packages/esm-bed-management-app/src/bed-administration/form/add-edit-bed.resource.ts

```diff
@@ -0,0 +1,102 @@
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import { type FetchResponse, openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import { type BedPostPayload, type BedTag } from '../../types';
+
+interface BedForm {
+  bedNumber: string;
+  bedType: string;
+  row: number;
+  column: number;
+  status: string;
+  locationUuid: string;
+  uuid?: string;
+}
+
+export async function saveBed({ bedPayload }: { bedPayload: BedPostPayload }): Promise<FetchResponse<BedForm>> {
+  const response = await openmrsFetch(`${restBaseUrl}/bed`, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: {
+      bedNumber: bedPayload.bedNumber,
+      bedType: bedPayload.bedType,
+      row: bedPayload.row,
+      column: bedPayload.column,
+      status: bedPayload.status,
+      locationUuid: bedPayload.locationUuid,
+    },
+  });
+
+  if (response.status === 201 && bedPayload.bedTag && bedPayload.bedTag.length > 0) {
```

Comment:
I'm not sure we need to check for a `20x` response status here. We could just wrap it an `try/catch` and `openmrsFetch` should bubble the error up if there is one.